components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -88,62 +80,68 @@ impl LocaleCanonicalizer<'_> {
 
         // languages_scripts_regions
         if locale.script.is_some() && locale.region.is_some() {
```

Comment:
Can you instead unpack it as:

```rust
match (locale.script, locale.region) {
    (Some(script), Some(region)) => {}
}
```

I'd really like us to avoid `unwraps` when not necessary.